process/block/metablock.go

```diff
@@ -2051,3 +1914,19 @@ func (mp *metaProcessor) GetBlockBodyFromPool(headerHandler data.HeaderHandler)
 
 	return block.Body(miniBlocks), nil
 }
+
+func (mp *metaProcessor) getLastCrossNotarizedHeaders() (map[uint32]data.HeaderHandler, error) {
```

Comment:
move this to block tracker.

Reply:
GetLastCrossNotarizedHeader from block tracker returns the last headers/hash for a given shard, you may not want all the time all the last headers for all the shards. Here the returned type is a map. I would not make another method GetLastCrossNotarizedHeaderForAllShards only for this case. I think that if one situation requires all the last notarized headers from all the shards it could be solved easily by iterate on each shard.

Reply:
BlockTracker would implement this function as well, with the same input and output. The thing is, all the data and processing inside the method is using only blockTracker data -> it can be done there.